src/components/input/Select.tsx

```diff
@@ -190,22 +190,19 @@ function SelectOption<T>({
     >
       <div
         className={classnames(
-          // Make items stretch so that all have the same height. This is
-          // important for multi-selects, where the checkbox actionable surface
-          // should span to the very edges of the option containing it.
-          'flex justify-between items-stretch',
```

Comment:
Truncating text did not properly work when nesting flex containers, so I simplified this to be a single flex container again, and making the checkbox for multi-selects be `self-stretch`.

This not only solves the problem, but also simplifies the overall layout making it more clear.